kitsune/search/v2/documents.py

```diff
@@ -111,43 +173,53 @@ class Index:
         using = config.DEFAULT_ES7_CONNECTION
 
     @classmethod
-    def prepare_locale(cls, instance):
-        return cls.get_question_instance(instance).locale
+    def prepare_question_creator(cls, instance):
+        user = cls.get_question_instance(instance).creator
```

Comment:
What's the value of using the class method `get_question_instance` and not directly use the `instance`? This add some confusion and  makes the code less readable.

Reply:
It's redefined in `AnswerDocument` to `return instance.question`, so these methods will work both when `instance` is of type `Question` and `Answer`. Agreed readability is quite bad, `AnswerDocument` could instead redefine `get_field_value` to call these methods with `instance.question` instead - that would also reduce the change of a later addition to these methods being buggy, because `instance` would always be of type `Question`.

Reply:
You can also add the `get_question_instance` as a class method only in the answer document and skip it from here.